util/settings/settings.go

```diff
@@ -134,10 +136,15 @@ type KustomizeSettings struct {
 
 func (ks *KustomizeSettings) GetOptions(source v1alpha1.ApplicationSource) (*v1alpha1.KustomizeOptions, error) {
 	binaryPath := ""
+	buildOptions := ks.BuildOptions
 	if source.Kustomize != nil && source.Kustomize.Version != "" {
 		for _, ver := range ks.Versions {
 			if ver.Name == source.Kustomize.Version {
 				binaryPath = ver.Path
+				// append version specific build options to global build options
```

Comment:
The options specified in `kustomize.buildOptions` is default version build options, not global options. So instead of appending, we should just replace version-specific build options.  If version specific build option is empty then we need to use empty build options.


Imagine the following use case. Everyone is org is using kustomize v3.9.x with `--load_restrictor none` build options.

In order to migrate to v4.x the admin would have to register new v4.0.x version with v4.0 specific  build options
( v4.x does not support `--load_restrictor none` ). Migrate all apps to v4.0.x and them change default to v4.0.x

Reply:
Good point.   With appending, there is really no way to change the default build options.

Reply:
That makes sense. I've updated the PR with the below logic

1. Use `kustomize.buildOptions` only for the default version and don't include it for version-specific options.
2. Use `kustomize.buildOptions.<version>` for version-specific options. If this field is absent, this version will have no build options.

Reply:
@alexmt The PR is updated, please take a look

Reply:
Hi guys,
Is this tested with `kustomize v.4.0.5`? 
I'm getting the following `kustomize version v4.0.5 is not registered` error. 

I checked that the `argocd-cm` is defined and the tool is in the specified path in the container as well.
Any ideas?

Reply:
Hi, could you try setting the path for v4.0.5 `kustomize.path.v4.0.5: <path>` in the `argocd-cm`. I think we get this error when there's no path specified for the kustomize version.

Reply:
Of course, I have correct path in configmap and the container has the tool under same path.
Updating my Argo installation from '1.8.7' to '2.0.1' made it work!